frontend/src/components/common/Resource/ScaleButton.tsx

```diff
@@ -163,7 +163,9 @@ function ScaleDialog(props: ScaleDialogProps) {
               <Fab
                 size="small"
                 color="primary"
-                onClick={() => setNumReplicas(numReplicas => numReplicas - 1)}
+                onClick={() =>
+                  setNumReplicas(numReplicas => (numReplicas <= 0 ? 0 : numReplicas - 1))
+                }
```

Comment:
You should also disable the button if numReplicas <= 0.

Reply:
done